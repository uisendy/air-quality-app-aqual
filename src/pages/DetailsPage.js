import React from 'react';
import { AiOutlineCloseCircle } from 'react-icons/ai';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';

const DetailsPage = () => {
  return (
    <div>
      <Link to={'/'}>
        <AiOutlineCloseCircle />
      </Link>
      <h2>New York</h2>
    </div>
  );
};

export default DetailsPage;
