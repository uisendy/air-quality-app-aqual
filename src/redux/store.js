import { configureStore } from '@reduxjs/toolkit';
import airQualityReducer from './airQualitySlice';
import searchLocationsReducer from './searchLocationsSlice';

const store = configureStore({
  reducer: {
    airQuality: airQualityReducer,
    locations: searchLocationsReducer,
  },
});

export default store;
