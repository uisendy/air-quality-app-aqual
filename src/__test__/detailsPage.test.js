import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import store from '../redux/store';
import DetailsPage from '../pages/DetailsPage';

afterEach(cleanup);
test('Check if Details page is rendering correctly', () => {
  const detailsPage = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Routes>
            <Route path="search/details" element={<DetailsPage />} />
          </Routes>
          ,
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(detailsPage).toMatchSnapshot();
});
