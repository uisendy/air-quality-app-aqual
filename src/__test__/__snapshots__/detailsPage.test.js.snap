// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Details page is rendering correctly 1`] = `","`;
